Fix tests with no-dh and no-ec

alyssais opened the pull request:
<!--
Thank you for your pull request. Please review these requirements:

Contributors guide: https://github.com/openssl/openssl/blob/master/CONTRIBUTING.md

Other than that, provide a description above this comment if there isn't one already

If this fixes a GitHub issue, make sure to have a line saying 'Fixes #XXXX' (without quotes) in the commit message.
-->

##### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->
- [x] tests are added or updated

alyssais committed:
Fix tests with no-dh and no-ec

paulidale reviewed (commented):
I would have expected to see a CI change/addition that exercises these.

alyssais committed:
Test combination of no-dh and no-ec

This has broken repeatedly.  It was fixed previously by
ddf8f1ce63 ("Ensure default supported groups works even with no-ec and
no-dh"), and I'm about to fix it again.  To make this the last time
this has to happen, test it for future contributions.

alyssais commented:
@t8m could you re-review?

openssl-machine commented:
This PR is in a state where it requires action by @openssl/committers but the last update was 31 days ago

slontis commented on test/threadstest.h:
```diff
@@ -69,12 +69,12 @@ static void *thread_run(void *arg)
 }
 
-static int wait_for_thread(thread_t thread)
+static ossl_unused int wait_for_thread(thread_t thread)
```
This is not resolved.

alyssais replied:
Sorry for the oversight. Now fixed.

slontis commented:
I tried your branch locally, and the tests fail.. 
>./config -Werror enable-fips --debug --strict-warnings

alyssais commented:
Passes for me with the same configuration.  Can you give more information?  What OS and architecture?  Logs?

nhorman commented:
This still fails:
```
./Configure --strict-warnings --debug enable-fips no-dh no-ec
...
make test
...
Test Summary Report
-------------------
80-test_ssl_new.t                      (Wstat: 4352 (exited 17) Tests: 31 Failed: 17)
  Failed tests:  1-11, 13, 15-18, 27
  Non-zero exit status: 17
Files=364, Tests=3226, 36 wallclock secs ( 3.47 usr  0.43 sys + 330.65 cusr 80.78 csys = 415.33 CPU)
Result: FAIL
make[2]: *** [Makefile:4090: run_tests] Error 1
make[2]: Leaving directory '/home/nhorman/git/openssl'
make[1]: *** [Makefile:4087: _tests] Error 2
make[1]: Leaving directory '/home/nhorman/git/openssl'
make: *** [Makefile:4085: tests] Error 2
```

openssl-machine commented:
This PR is in a state where it requires action by @openssl/committers but the last update was 30 days ago

openssl-machine commented:
This PR is in a state where it requires action by @openssl/committers but the last update was 61 days ago

openssl-machine commented:
This PR is in a state where it requires action by @openssl/committers but the last update was 92 days ago

beldmit reviewed (changes requested):
Needs rebase, will approve after that
